 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>

<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$

**Refrigerant 134a enters an adiabatic compressor as saturated vapor at 120 kPa at a rate of 0.3 m³/min and exits at 1 MPa. If the isentropic efficiency of the compressor is 80%, determine:**
1. the temperature of the refrigerant at the exit of the compressor and
1. the power input, in kW.       


### Given
<ol> 
 $\text{R-134a} \left| \begin{array}{ll} 
  P_{1} = 120\, \ce{kPa} \\[0pt]
  x_{1} = \text{sat vapor} \\[0pt]
  \dot V_{1} = 0.3\, \ce{m^3/min}
 \end{array}\right|
 \xrightarrow{\eta_{II} = 80\%}
 $
 $\left| \begin{array}{ll} 
  \\[0pt]
  P_{2} = 1\, \ce{MPa} \\[0pt]
  \\[0pt]
 \end{array}\right|
 $
 

In [1]:
using Unitful       # Unitful units package

In [2]:
# Given 
P1 = 120u"kPa"  
x1 = 1
V̇1 = 0.3u"m^3/minute"
ηii = 0.8;

### Find 
1. the temperature of the refrigerant at the exit of the compressor and
1. the power input, in kW. 

### Get Organized  and Attack
We are given State 1 and an isentropic efficiency $(\eta_{\ce{II}})$, so we  
1. determine the isentropic work from State 1 to State 2 ($\dot W_s$),  
2. use $\eta_{\ce{II}}$ to calculate the actual work ($\dot W_a$), and  
3. use $\dot W_a$ to get State 2, which gives us $T_2$

Conservation of mass is trivial, so we can jump right into conservation of energy. 

#### State 1 is GIven

In [3]:
# STATE 1
# P1 = 120 kPa, x1 = 1 ⋮ need s1, h1

s1 = 0.9354u"kJ/(kg*K)"
h1 = 233.86u"kJ/kg";

#### Determine Isentropic State 2s  
    
$$ \left. \begin{array}{cl}
  \dot E_{in} &-& \dot E_{out} &=& \frac{d}{dt} E_{sys} \\[6pt]
  \dot m h_1 + \dot m w_{s,in} &-& \dot m h_{2s} &=& 0 \\[6pt]
\end{array} \right\} \quad
 w_{s,in} =  h_{2s} - h_1  \\[6pt]
$$  

In [4]:
# STATE 2S
# P2s = 1 MPa, s2 = s1 ⋮ superheated ⋮ need h2s ⋮ must interpolate
# (h2s - h_40°C)/(h_50°C - h_40°C) = (s2 - s_40°C)/(s_50°C - s_40°C)

s2 = s1
s_40°C = 0.9066u"kJ/(kg*K)"
s_50°C = 0.9428u"kJ/(kg*K)"
h_40°C = 268.68u"kJ/kg"
h_50°C = 280.19u"kJ/kg"

h2s = h_40°C + (h_50°C - h_40°C)*(s2 - s_40°C)/(s_50°C - s_40°C);

# println("h2s = ",h2s)

#### Calculate Actual Work 
Remember, the isentropic work is the best-case scenario. Compressors are work-consuming devices, so we expect $ w_s < w_a,$ therefore,  

$$\displaystyle \eta_{\ce{II}} = \frac{w_{s,in}}{w_{a,in}}$$  

$$\displaystyle w_{a,in} = \frac{w_{s,in}}{\eta_{\ce{II}}} = \frac{h_{2s} - h_1}{\eta_{\ce{II}}} $$  

$$$$

$$ \color{red} \dot W_{a,in} = \dot m w_{a,in} = \frac{\dot V}{v}w_{a,in}$$

In [5]:
# Work_actual
v1 = 0.1614u"m^3/kg"    # from saturation table

w_a = (h2s - h1)/ηii
Ẇ_a = w_a * V̇1/v1 |> u"kW"

println("Ẇ_actual = ",Ẇ_a)

Ẇ_actual = 1.7029556641815053 kW


<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize \dot W_{in} = 1.703\, \ce{kW} $$
</div></div>

#### Get Actual State 2  for $T_2$

With actual work in hand, it's a small matter to resolve State 2.  
$$$$  
$$ {w_{a,in}} = h_{2a} - h_1 $$

$$$$

 $$\text{R-134a} \left| \begin{array}{ll} 
  P_{1} = 120\, \ce{kPa} \\[0pt]
  x_{1} = \text{sat vapor} \\[0pt]
  \hline 
  \text{get } s_1,\ h_1 \\[0pt] 
  \\[0pt]
 \end{array}\right|
 \longrightarrow
 \left| \begin{array}{ll} 
   P_{2s} = 1\, \ce{MPa} \\[0pt]
   s_2 = s_1 \\[0pt]
   \hline
   \text{get } h_{2s} \\[0pt]
   \text{compute } w_s,\ w_a \\[0pt]   
 \end{array}\right|
  \longrightarrow
 \left| \begin{array}{ll} 
   P_{2a} = 1\, \ce{MPa} \\[0pt]
   h_{2a} = h_1 + {w_{a,in}} \\[0pt]
   \hline
   \color{red}\text{get } T_{2a} \\[0pt]
   \\[0pt]   
 \end{array}\right|
 $$

In [6]:
# STATE 2
# P2 = 1 MPa, h2a = h1 + w_a 

h2a = h1 + w_a
println("h2a = ",h2a)

h2a = 288.831408839779 kJ kg^-1


In [7]:
#  need T2 ⋮ must interpolate
# (T2 - 50°C)/(60°C - 50°C) = (h2a - h_50°C)/(h_60°C - h_50°C)

h_50°C = 280.19u"kJ/kg"
h_60°C = 291.36u"kJ/kg"

T2 = 50u"°C" + (h2a - h_50°C)/(h_60°C - h_50°C)*10u"K"

println("T2 = ",T2 |> u"°C")

T2 = 57.736265747340155 °C


<div class="panel-success"><div class="panel-heading"><div class="badge"></div> $$\ \normalsize T_2 = 57.74\, \ce{°C} $$
</div></div>

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>